modules/test/playwright/tests/osb-faro-web/utils/data-source.ts

```diff
@@ -24,3 +48,63 @@ export async function createDataSource(page) {
 
 	await page.getByRole('link', {name: 'Done'}).click();
 }
+
+export async function renameDataSource({
+	newDataSourceName,
+	oldDataSourceName,
+	page,
+}: {
+	newDataSourceName: string;
+	oldDataSourceName: string;
+	page: Page;
+}) {
+	await searchDataSource({
+		clickOnDataSourceName: true,
+		dataSourceName: oldDataSourceName,
+		page,
+	});
+
+	await page.getByRole('button', {name: 'Edit'}).click();
+
+	await page.locator('#dataSourceName').click();
+
+	await page.locator('#dataSourceName').fill(newDataSourceName);
+
+	await page
+		.getByRole('button', {name: 'Submit'})
+		.waitFor({state: 'visible'});
+
+	await page.getByRole('button', {name: 'Submit'}).click();
+
+	await expect(
+		page
+			.getByRole('heading', {exact: true, name: newDataSourceName})
+			.locator('span')
+	).toBeVisible();
+}
+
+export async function searchDataSource({
```

Comment:
Can you rename from `searchDataSource` to `findDataSource` and this should be your return:

`return await page.getByRole('link', {exact: true, name: dataSourceName})`
			
And then, do something like:

```
const dataSource = await findDataSource({dataSourceName, page});

if (dataSource) {
   await dataSource.click();
}
```

Reply:
done